app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -0,0 +1,88 @@
+package ar.com.wolox.android.example.ui.login;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.widget.EditText;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import ar.com.wolox.wolmo.core.presenter.BasePresenter;
+
+/**
+ * LoginPresenter
+ */
+
+public class LoginPresenter extends BasePresenter<ILoginView> {
+
+    final String emailReq = "Email is required";
+    final String passReq = "Password is required";
+    final String invEmail = "Invalid Email";
+    final String emailKey = "ar.com.wolox.android.example.emailCredential";
+    final String passKey = "ar.com.wolox.android.example.passCredential";
+
+    @Inject
+    LoginPresenter() {}
+
+    /**
+     * @param email EditText with the User Email
+     * @param pass EditText with the User Password
+     * @param activity Activity of LoginFragment
+     * @param context Context of LoginFragment
+     */
+    public void onLoginButtonClicked(EditText email, EditText pass, Activity activity, Context context) {
+
+        if (email.getText().toString().isEmpty() && pass.getText().toString().isEmpty()) {
+            getView().setEmptyEmailPass(emailReq, passReq);
```

Comment:
We could only call `setEmptyEmailPass()` with no arguments and let the view decide the message to show since that is UI stuff.